tests/tests/browser/head.js

```diff
@@ -232,3 +299,298 @@ async function withAddon(callback) {
+  }
+}
+
+/**
+ * Performs a step in a mock update.  This function and the other update-related
+ * functions are adapted from `runAboutDialogUpdateTest` here:
+ * https://searchfox.org/mozilla-central/source/toolkit/mozapps/update/tests/browser/head.js
+ */
+async function processUpdateStep(step) {
+  if (typeof step == "function") {
+    step();
+    return;
+  }
+
+  const { panelId, checkActiveUpdate, continueFile, downloadInfo } = step;
+  if (checkActiveUpdate) {
+    await TestUtils.waitForCondition(
+      () => gUpdateManager.activeUpdate,
+      "Waiting for active update"
+    );
+    Assert.ok(
+      !!gUpdateManager.activeUpdate,
+      "There should be an active update"
+    );
+    Assert.equal(
+      gUpdateManager.activeUpdate.state,
+      checkActiveUpdate.state,
+      "The active update state should equal " + checkActiveUpdate.state
+    );
+  } else {
+    Assert.ok(
+      !gUpdateManager.activeUpdate,
+      "There should not be an active update"
+    );
+  }
+
+  if (panelId == "downloading") {
+    for (let i = 0; i < downloadInfo.length; ++i) {
+      let data = downloadInfo[i];
+      // The About Dialog tests always specify a continue file.
+      await continueFileHandler(continueFile);
+      let patch = getPatchOfType(data.patchType);
+      // The update is removed early when the last download fails so check
+      // that there is a patch before proceeding.
+      let isLastPatch = i == downloadInfo.length - 1;
+      if (!isLastPatch || patch) {
+        let resultName = data.bitsResult ? "bitsResult" : "internalResult";
+        patch.QueryInterface(Ci.nsIWritablePropertyBag);
+        await TestUtils.waitForCondition(
+          () => patch.getProperty(resultName) == data[resultName],
+          "Waiting for expected patch property " +
+            resultName +
+            " value: " +
+            data[resultName],
+          undefined,
+          200
+        ).catch(e => {
+          // Instead of throwing let the check below fail the test so the
+          // property value and the expected property value is printed in
+          // the log.
+          logTestInfo(e);
+        });
+        Assert.equal(
+          patch.getProperty(resultName),
+          data[resultName],
+          "The patch property " +
+            resultName +
+            " value should equal " +
+            data[resultName]
+        );
+      }
+    }
+  } else if (continueFile) {
+    await continueFileHandler(continueFile);
+  }
+}
+
+/**
+ * Starts a search and asserts that the second result is a tip.
+ *
+ * @param {string} searchString
+ *   The search string.
+ * @return {[result, element]}
+ *   The result and its element in the DOM.
+ */
+async function awaitTip(searchString) {
+  let context = await UrlbarTestUtils.promiseAutocompleteResultPopup({
+    window,
+    value: searchString,
+    waitForFocus,
+    fireInputEvent: true,
+  });
+  Assert.ok(context.results.length >= 2);
+  let result = context.results[1];
```

Comment:
our wrapper here is getDetailsOfResultAt and then you don't even need the later call to waitForAutocompleteResultAt, just use .element

Reply:
The result itself isn't available from `getDetailsOfResultAt` afaict. We should just get rid of `getDetailsOfResultAt`. It was helpful when it was abstracting away the differences between the two implementations of the urlbar, but now it's just annoying.